### PR TITLE
docs: update retainer docs

### PR DESCRIPTION
**Pull request by lafirest**

### CODE REVIEW

File: en_US/advanced/retained.md [hunk 1]
```diff
@@ -17,13 +14,32 @@ With retained messages, new subscribers can immediately get the latest status wi
...
 subscribers may need get the latest reading immediately after subscribed, without having to wait for the next publish.
 
-In EMQX, the retainer is enabled by default. To turn it off, you can set `mqtt.retain_available = false` globally for the entire broker
-or `zones.$name.mqtt.retain_available = false` to turn it off only for a given config group.
+EMQX enables the capability and service of retain messages by default. You can modify `mqtt.retain_available` to `false` in `etc/emqx.conf` to disable the ability to retain messages.
+In this way, the client will be prohibited from sending PUBLISH packets with the Retain flag set to 1. Otherwise, the client will receive a DISCONNECT packet with a reason code of 0x9A (Retain not supported).
+
+The service stores and manages retained messages sent by clients and sends them to the corresponding subscribers.
+
+## The Service
```
Comment (HJianBo) [inline en_US/advanced/retained.md:22]: ## Quick Start

Comment (hjianbo) [inline en_US/advanced/retained.md]:
```suggestion
## Quick Start
```

File: en_US/advanced/retained.md [hunk 2]
```diff
@@ -17,13 +14,32 @@ With retained messages, new subscribers can immediately get the latest status wi
...
+| Expire       | duration |                          | ０     | The expiration time of retaining message, and 0 means never expire. If the message expiration interval is set in the PUBLISH packet, the message expiration interval in the PUBLISH packet shall prevail. |
+| Clean Interval  | duration |                          | ０     | Interval to clean up expired messages. |
+
+<!---
 
-::: warning
-If EMQX receives a retained message when the feature is turned off,
-it will reply a DISCONNECT message with reason code 0x9A (retained message is not supported).
-:::
+TODO wait new limiter
```
Comment (hjianbo) [inline en_US/advanced/retained.md]:
```suggestion

```

File: zh_CN/advanced/retained.md [hunk 3]
```diff
@@ -10,13 +8,34 @@ MQTT 服务端收到 Retain 标志为 1 的 PUBLISH 报文时，会将该报文
 当客户端建立订阅时，如果服务端存在主题匹配的保留消息，则这些保留消息将被立即发送给该客户端。
 借助保留消息，新的订阅者能够立即获取最近的状态，而不需要等待无法预期的时间，这在很多场景下是非常重要的。
 
-EMQX 默认开启保留消息的功能，可以在 `etc/emqx.conf` 中修改 `mqtt.retain_available` 为 `false` 以禁用保留消息功能。
+EMQX 默认开启保留消息的能力和服务，可以在 `etc/emqx.conf` 中修改 `mqtt.retain_available` 为 `false` 来关闭保留消息的能力，
+这样客户端将被禁止发送 Retain 标志为 1 的 PUBLISH 报文，否则，客户端将会收到原因码为 0x9A（不支持保留消息）的 DISCONNECT 报文。
+
+保留消息的服务会存储和管理客户端发送的保留消息，并发送给相应的订阅者。
+
+## 保留消息服务
```
Comment (hjianbo) [inline zh_CN/advanced/retained.md]:
```suggestion
## 快速开始
```

File: zh_CN/advanced/retained.md [hunk 4]
```diff
@@ -10,13 +8,34 @@ MQTT 服务端收到 Retain 标志为 1 的 PUBLISH 报文时，会将该报文
...
+| 有效期       | duration |       | ０     | 保留消息的过期时间，0 表示永不过期。如果 PUBLISH 报文中设置了消息过期间隔，那么以 PUBLISH 报文中的消息过期间隔为准。 |
+| 定时清理       | duration |       | ０     | 清理回收过期消息的间隔时间。 |
 
-::: warning
-如果 EMQX 在保留消息功能被禁用的情况下依然收到了保留消息，那么将返回原因码为 0x9A（不支持保留消息）的 DISCONNECT 报文。
-:::
 
 <!---
+
+TODO wait new limiter
```
Comment (hjianbo) [inline zh_CN/advanced/retained.md]:
```suggestion

```

File: en_US/advanced/retained.md (hunk 2 above)
Comment (JimMoen) [reply en_US/advanced/retained.md]: In comment, it will be ignored.

File: zh_CN/advanced/retained.md (hunk 4 above)
Comment (JimMoen) [reply zh_CN/advanced/retained.md]: same.

File: en_US/advanced/retained.md [hunk 5]
```diff
@@ -17,13 +14,32 @@ With retained messages, new subscribers can immediately get the latest status wi
 which is very important in many scenarios. For instance sensors may only publish readings every few minutes, but the
 subscribers may need get the latest reading immediately after subscribed, without having to wait for the next publish.
 
-In EMQX, the retainer is enabled by default. To turn it off, you can set `mqtt.retain_available = false` globally for the entire broker
-or `zones.$name.mqtt.retain_available = false` to turn it off only for a given config group.
+EMQX enables the capability and service of retain messages by default. You can modify `mqtt.retain_available` to `false` in `etc/emqx.conf` to disable the ability to retain messages.
```
Comment (JimMoen) [inline en_US/advanced/retained.md:17]: Maybe not `mqtt`**DOT**`retain_available` in hocon style?

File: en_US/advanced/retained.md [hunk 6]
```diff
@@ -17,13 +14,32 @@ With retained messages, new subscribers can immediately get the latest status wi
...
+| ------------------------------ | -------- | ------------------------ | ------ | ------------------------------------------------------------ |
+| Storage         | enum     | `ram`, `disc` | ram |ram: only stored in memory; <br /> disc: stored in memory and hard disk. |
+| Max Retained Messages | integer  | \>= 0                    | 0      | The maximum number of retained messages, and 0 means no limit. After the number of retained messages exceeds the maximum limit, you can replace the existing retained messages, but cannot store retained messages for new topics. |
+| Max Payload Size      | bytesize |                          | 1MB    | Retain the maximum Payload value of the message. After the Payload value exceeds the maximum value, the EMQX broker will treat the retained reserved message as a normal message. |
+| Expire       | duration |                          | ０     | The expiration time of retaining message, and 0 means never expire. If the message expiration interval is set in the PUBLISH packet, the message expiration interval in the PUBLISH packet shall prevail. |
```
Comment (JimMoen) [inline en_US/advanced/retained.md]: Use half-angle characters, please.

**Review by hjianbo — APPROVED** [review]

**Pull request merged**